frontend/src/components/Board/BuyParts/DirectStores.js

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react'
+
+import { array, number } from 'prop-types'
+import DigikeyData from '1-click-bom-minimal/lib/data/digikey.json'
+import FarnellData from '1-click-bom-minimal/lib/data/farnell.json'
+import countriesData from '1-click-bom-minimal/lib/data/countries.json'
+
+const DirectStores = ({ items, multiplier }) => {
+  const [countryCode, setCountryCode] = useState('Other')
+  const [digikeyParts, setDigikeyParts] = useState([])
+  const [farnellParts, setFarnellParts] = useState([])
+  const [newarkParts, setNewarkParts] = useState([])
+
+  const getParts = retailer =>
+    items
+      .filter(part => retailer in part.retailers && part.retailers[retailer] !== '')
+      .map(part => ({
+        sku: part.retailers[retailer],
+        reference: part.reference,
+        quantity: Math.ceil(multiplier * part.quantity),
+      }))
+
+  const getLocation = async () => {
+    const usedCountryCodes = Object.keys(countriesData).map(
+      key => countriesData[key],
+    )
+    const freegeoipEndpoint = 'https://freegeoip.kitspace.org'
+
+    try {
+      const res = await fetch(freegeoipEndpoint)
+      const body = await res.json()
+      const { country_code: code } = body
+      if (code === 'GB') {
+        return 'UK'
+      }
+      if (usedCountryCodes.indexOf(code) < 0) {
+        return 'Other'
+      }
+      return code
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err)
```

Comment:
I think we should make global exceptions for `console.error` and `console.warn`.